public/app/plugins/datasource/prometheus/completer.ts

```diff
@@ -5,32 +5,46 @@ export class PromCompleter {
   labelQueryCache: any;
   labelNameCache: any;
   labelValueCache: any;
+  templateVariableCompletions: any;
 
   identifierRegexps = [/\[/, /[a-zA-Z0-9_:]/];
 
-  constructor(private datasource: PrometheusDatasource) {
+  constructor(private datasource: PrometheusDatasource, private templateSrv) {
     this.labelQueryCache = {};
     this.labelNameCache = {};
     this.labelValueCache = {};
+    this.templateVariableCompletions = this.templateSrv.variables.map(function (variable) {
+      return {
+        caption: '$' + variable.name,
+        value: '$' + variable.name,
```

Comment:
Looks good, but what do you think about  `[[variable]]` style? I'm trying to add this, but have a little trouble on this. It works well inside a string, but when I add it in the other place, it adds two extra brackets, from behaviour rule for `[]`. I use this code:
```js
let templateVariableBracketCompletions = this.templateSrv.variables.map(function (variable) {
      return {
        caption: '[[' + variable.name + ']]',
        value: '[[' + variable.name + ']]',
        meta: "variable.other",
        score: Number.MAX_VALUE
      };
    });
```

Reply:
Remove last `]` character will work in expression part.
I don't think it is good way, but behaviour rule was executed before displaying variable candidate, fixing the behaviour is impossible...